client/extensions/woocommerce/state/sites/customers/selectors.js

```diff
@@ -1,86 +1,59 @@
-/**
- * External dependencies
- */
-
-import { filter, get } from 'lodash';
-
 /**
  * Internal dependencies
  */
 import { getSelectedSiteId } from 'state/ui/selectors';
 
+function inFlight( state, siteId, searchTerm ) {
+	return state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers?.isSearching?.[ searchTerm ];
```

Comment:
When I see long `?.` chains like this, it suggests a question: are the referenced fields really optional, or are they always present?

I understand that `state.extensions.woocommerce` doesn't need to be present when the dynamic Woo reducer is not loaded in the store. It's a good practice for the selector to return answer that means "value is unknown" in such a case.

But on the `customers` state, the `isSearching` field is always present. The reducer is a result of `combineReducers( { isSearching, items, queries } )`.

Similarly, when the state has a record for `siteId`, the `customers` field is guaranteed to be present. For similar reasons: because of `combineReducers`.

It's therefore safe to write `[ siteId ]?.customers.isSearching[ searchTerm ]`

To say the same idea with different words: even though this code is not written in TypeScript, we can still _think in types_ when writing it. The `Customers` type has a mandatory field `isSearching` and it's safe to access it with `customers.isSearching`.

Reply:
That makes sense, yes. I was coming at this from the perspective of "translate what `get` did", but since I'm refactoring elsewhere I may as well refactor here too.

Reply:
@jsnajdr Done, thanks! 👍